src/Aspire.Dashboard/Components/Controls/LogViewer.razor.cs

```diff
@@ -1,73 +1,74 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
-using System.Collections.Concurrent;
 using Aspire.Dashboard.ConsoleLogs;
 using Aspire.Dashboard.Model;
+using Aspire.Dashboard.Utils;
 using Microsoft.JSInterop;
 
 namespace Aspire.Dashboard.Components;
-public partial class LogViewer
+
+/// <summary>
+/// A log viewing UI component that shows a live view of a log, with syntax highlighting and automatic scrolling.
+/// </summary>
+public sealed partial class LogViewer
 {
-    private readonly ConcurrentQueue<IEnumerable<LogEntry>> _preRenderQueue = new();
-    private bool _renderComplete;
+    private readonly TaskCompletionSource _whenDomReady = new();
```

Comment:
nit: At least in aspnetcore repo we add Tcs onto `TaskCompleteSource` variable names.

Reply:
It's not very consistent in this repo:

```
src\Aspire.Dashboard\Components\Controls\LogViewer.razor.cs(16):private readonly TaskCompletionSource _whenDomReady = new();
src\Aspire.Dashboard\Components\Pages\ConsoleLogs.razor.cs(24):private readonly TaskCompletionSource _whenFirstRenderComplete = new();
src\Aspire.Hosting\Dashboard\DockerContainerLogSource.cs(36):var tcs = new TaskCompletionSource();
src\Aspire.Hosting\Dashboard\DockerContainerLogSource.cs(66):async Task WaitForExit(TaskCompletionSource tcs, CancellationTokenRegistration ctr)
src\Aspire.Hosting\Dcp\Process\ProcessUtil.cs(80):var processLifetimeTcs = new TaskCompletionSource<ProcessResult>();
src\Microsoft.Extensions.ServiceDiscovery\Http\HttpServiceEndPointResolver.cs(166):private TaskCompletionSource? _onDisposed;
src\Microsoft.Extensions.ServiceDiscovery\ServiceEndPointResolver.cs(338):TaskCompletionSource completion = new(TaskCreationOptions.RunContinuationsAsynchronously);
src\Microsoft.Extensions.ServiceDiscovery\ServiceEndPointResolver.cs(349):changeTokenRegistration = changeToken.RegisterChangeCallback(static state => ((TaskCompletionSource)state!).TrySetResult(), completion);
src\Microsoft.Extensions.ServiceDiscovery\ServiceEndPointResolver.cs(354):pollPeriodRegistration = timerCancellation.Token.UnsafeRegister(static state => ((TaskCompletionSource)state!).TrySetResult(), completion);
src\Microsoft.Extensions.ServiceDiscovery\ServiceEndPointResolver.cs(359):cancellationRegistration = cancellationToken.UnsafeRegister(static state => ((TaskCompletionSource)state!).TrySetResult(), completion);
src\Microsoft.Extensions.ServiceDiscovery\ServiceEndPointResolverRegistry.cs(162):private TaskCompletionSource? _onDisposed;
```

I like "when" for this kind of coordination as the name generally ends up being more indicative of why the tcs exists.